core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -759,6 +758,68 @@ protected RexNode removeCorrelationExpr(
     }
 
     RelNode newRel = relBuilder.build();
+    // Special case where the group by is static (i.e., a select clause with aggregation functions
+    // but without group by).
+    if (rel.getGroupType() == Aggregate.Group.SIMPLE
+        && rel.getGroupSet().isEmpty()
+        && !frame.corDefOutputs.isEmpty()) {
+      final Pair<CorrelationId, Frame> outerFramePair = requireNonNull(this.frameStack.peek());
+      final Frame outFrame = outerFramePair.right;
+      RexBuilder rexBuilder = relBuilder.getRexBuilder();
+
+      int groupSize = (int) corDefOutputs.keySet().stream()
+          .filter(a -> a.corr.equals(outerFramePair.left))
+          .count();
+      List<RelDataTypeField> newRelFields = newRel.getRowType().getFieldList();
+      ImmutableBitSet.Builder corFieldBuilder = ImmutableBitSet.builder();
+      final Map<Integer, RexNode> specificRexNodeMap = new HashMap<>();
+      final List<RexNode> conditions = new ArrayList<>();
+      for (Map.Entry<CorDef, Integer> corDefOutput : corDefOutputs.entrySet()) {
+        if (corDefOutput.getKey().corr.equals(outerFramePair.left)) {
+          int newIdx = requireNonNull(outFrame.oldToNewOutputs.get(corDefOutput.getKey().field));
+          corFieldBuilder.set(newIdx);
+
+          RelDataType type = outFrame.r.getRowType().getFieldList().get(newIdx).getType();
+          RexNode left = new RexInputRef(corFieldBuilder.cardinality() - 1, type);
+          specificRexNodeMap.put(corDefOutput.getValue() + groupSize, left);
+          conditions.add(
+              relBuilder.isNotDistinctFrom(left,
+                  new RexInputRef(corDefOutput.getValue() + groupSize,
+                      newRelFields.get(corDefOutput.getValue()).getType())));
+        }
+      }
+
+      ImmutableBitSet groupSet = corFieldBuilder.build();
+      final RelNode join = relBuilder.push(outFrame.r)
+          .aggregate(relBuilder.groupKey(groupSet))
+          .push(newRel)
+          .join(JoinRelType.LEFT, conditions).build();
+
+      for (int i1 = 0; i1 < rel.getAggCallList().size(); i1++) {
+        AggregateCall aggCall = rel.getAggCallList().get(i1);
+        if (aggCall.getAggregation() instanceof SqlCountAggFunction) {
```

Comment:
does the argument to count matter? i.e., `*`?
the paper seemed to imply it does